monai/deploy/operators/monai_seg_inference_operator.py

```diff
@@ -290,23 +294,39 @@ def verify_suffix(self, filename: Union[Sequence[str], str]) -> bool:
         return True
 
     def read(self, data: Union[Sequence[str], str], **kwargs) -> Union[Sequence[Any], Any]:
-        # Really does not have anything to do.
-        return self.input_image.asnumpy()
+        # Really does not have anything to do. Simply return the Image object
+        return self.input_image
 
     def get_data(self, input_image):
         """Extracts data array and meta data from loaded image and return them.
 
         This function returns two objects, first is numpy array of image data, second is dict of meta data.
         It constructs `affine`, `original_affine`, and `spatial_shape` and stores them in meta dict.
-        A single image is loaded with a single set of metadata as of now."""
+        A single image is loaded with a single set of metadata as of now.
+
+        The App SDK Image asnumpy function is expected to return a NumPy array of index order `DHW`.
+        This is because in the DICOM serie to volume operator pydicom Dataset pixel_array is used to
+        to get per instance pixel NumPy array, with index order of `HW`. When all instances are stacked,
+        along the first axis, the Image NumPy array's index order is `DHW`. ITK array_view_from_image
+        and SimpleITK GetArrayViewFromImage also returns a Numpy array with index order of `DHW`.
+        This NumPy array is then transposed to be consistent with the NumPy array from NibabelReader,
+        which loads NIfTI image and gets NumPy array using the image's get_fdata() function.
```

Comment:
Can you please mention in the comments what is the final order/format of the pixel data that get_data returns ?

Reply:
Added comments on the expected/normalized numpy array index order from get_data().